### ISSUE TITLE
generated code causes -Wextra-semi-stmt warning with YY_DO_BEFORE_ACTION

### ISSUE THREAD

**Issue by seanm** [westes/flex#416]
(I don't know/use flex really, so forgive any vagueness/inaccuracy.)

The CMake project uses some flex generated code. Building CMake with clang's -Wextra-semi-stmt warning gives warnings like:

```
CMake/Source/LexerParser/cmDependsJavaLexer.cxx:1176:22: warning: empty expression statement has no effect; remove unnecessary ';' to silence this warning [-Wextra-semi-stmt]
                YY_DO_BEFORE_ACTION;
                                   ^
```

A fix would be to change the YY_DO_BEFORE_ACTION macro to a `do {...} while (0)` form, so that a trailing semi is actually required at invocation.

**Comment by seanm** [comment]
@westes would my suggested change be ok?  I could make a patch...

**Comment by westes** [comment]
fixed by #576.
